openwisp_controller/config/tests/test_handlers.py

```diff
@@ -0,0 +1,30 @@
+from unittest.mock import patch
+
+from django.test import TestCase
+
+from openwisp_users.tests.utils import TestOrganizationMixin
+
+from .. import tasks
+
+
+class TestHandlers(TestOrganizationMixin, TestCase):
+    @patch.object(tasks.invalidate_device_checksum_view_cache, 'delay')
+    def test_organization_disabled_handler(self, mocked_task):
+        with self.subTest('Test task not executed on creating active orgs'):
+            org = self._create_org()
+            mocked_task.assert_not_called()
+
+        with self.subTest('Test task executed on changing active to inactive org'):
+            org.is_active = False
+            org.save()
+            mocked_task.assert_called_once()
+
+        mocked_task.reset_mock()
+        with self.subTest('Test task not executed on saving inactive org'):
+            org.name = 'Changed named'
+            org.save()
+            mocked_task.assert_not_called()
+
+        with self.subTest('Test task not executed on creating inactive org'):
+            self._create_org(is_active=False, name='inactive', slug='inactive')
+            mocked_task.assert_not_called()
```

Comment:
is the task called when the organization goes from inactive to active?

Reply:
No, it is not called when the organization changes from inactive to active.

Reply:
In my testing, I found that this function does not cache 404 response. Hence, there is no need to invalidate the cache. 

https://github.com/openwisp/openwisp-controller/blob/87a301a1ee40c38d8aba3daecf6f2c2581f4c5f0/openwisp_controller/config/controller/views.py#L149-L155

This gives me concern that devices of disabled organizations will keep generating queries for no good reason.

Reply:
> This gives me concern that devices of disabled organizations will keep generating database queries for no good reason.

It's a legitimate concern, however, we introduced a mechanism in [openwisp-config to stop the agent when 404 is received](https://github.com/openwisp/openwisp-config/issues/160), so over time, with the up to date version of the agents it will not be an issue.

We'll have to take care of the monitoring too: https://github.com/openwisp/openwrt-openwisp-monitoring/issues/105 .